src/execution.rs

```diff
@@ -320,16 +411,51 @@ impl Stanza {
                 shorthands: shorthands,
             };
             for statement in &self.statements {
-                statement
-                    .execute(&mut exec)
-                    .with_context(|| format!("Executing {}", statement))?;
+                statement.execute(&mut exec).or_else(|e| {
+                    Err(e).with_context(|| {
+                        let node = mat
+                            .captures
+                            .iter()
+                            .find(|c| c.index as usize == self.full_match_file_capture_index)
+                            .unwrap()
+                            .node;
+                        format!(
+                            "While executing statement {}\n{}\nin stanza\n{}\nmatching ({}) node\n{}",
+                            statement,
+                            Excerpt::from_source(tsg_path, tsg_source, statement.location()),
+                            Excerpt::from_source(tsg_path, tsg_source, self.location),
+                            node.kind(),
+                            Excerpt::from_source(
+                                source_path,
+                                source,
+                                Location::from(node.range().start_point)
+                            )
+                        )
+                    })
+                })?;
             }
         }
         Ok(())
     }
 }
 
 impl Statement {
+    pub fn location(&self) -> Location {
+        match self {
+            Statement::DeclareImmutable(s) => s.location,
+            Statement::DeclareMutable(s) => s.location,
+            Statement::Assign(s) => s.location,
+            Statement::CreateGraphNode(s) => s.location,
+            Statement::AddGraphNodeAttribute(s) => s.location,
+            Statement::CreateEdge(s) => s.location,
+            Statement::AddEdgeAttribute(s) => s.location,
+            Statement::Scan(s) => s.location,
+            Statement::Print(s) => s.location,
+            Statement::If(s) => s.location,
+            Statement::ForIn(s) => s.location,
+        }
+    }
```

Comment:
I feel like there ought to be a better way to do this sort of thing, but if there is, I don't know it.

Reply:
I don't know either.

Reply:
Then at least I'm in good company!